projects/igniteui-angular/src/lib/banner/banner.component.ts

```diff
@@ -195,6 +233,8 @@ export class IgxBannerComponent implements IToggleView {
     @ContentChild(IgxBannerActionsDirective)
     private _bannerActionTemplate: IgxBannerActionsDirective;
 
+    private _expanded: boolean = false;
+    private _isProgrammaticExpanded = false;
```

Comment:
can we rename this to something like `shouldFireEvent` or `fireEvent` or anything like this.